netlogo-gui/src/main/app/FileManager.scala

```diff
@@ -0,0 +1,403 @@
+// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo
+
+package org.nlogo.app
+
+import java.net.{ URI, URISyntaxException }
+import java.io.{ File, IOException }
+import java.nio.file.Paths
+import java.awt.{ Component, Container, FileDialog => AWTFileDialog }
+import javax.swing.{ Action, JOptionPane }, Action.{ ACCELERATOR_KEY }
+
+import org.nlogo.core.{ I18N, Model }
+import org.nlogo.api.{ Exceptions, ModelLoader, ModelReader, ModelType, Version }, ModelReader.{ emptyModelPath, modelSuffix }
+import org.nlogo.app.common.{ Actions, ExceptionCatchingAction }, Actions.Ellipsis
+import org.nlogo.app.tools.{ ModelsLibraryDialog, NetLogoWebSaver }
+import org.nlogo.awt.{ Hierarchy, UserCancelException }
+import org.nlogo.swing.{ FileDialog, ModalProgressTask, OptionDialog, UserAction }
+import org.nlogo.workspace.{ AbstractWorkspaceScala, OpenModel, SaveModel, SaveModelAs }
+import org.nlogo.window.{ BackgroundFileController, Events, FileController, ReconfigureWorkspaceUI },
+  Events.{ AboutToQuitEvent, LoadModelEvent, ModelSavedEvent, OpenModelEvent }
+import org.nlogo.fileformat.ModelConversion
+
+import scala.util.Try
+
+object FileManager {
+  class NewAction(manager: FileManager, parent: Container)
+  extends ExceptionCatchingAction(I18N.gui.get("menu.file.new"), parent) {
+    putValue(UserAction.ActionCategoryKey, UserAction.FileCategory)
+    putValue(UserAction.ActionGroupKey,    UserAction.FileOpenGroup)
+    putValue(UserAction.ActionRankKey,     Double.box(1))
+    putValue(ACCELERATOR_KEY,              UserAction.KeyBindings.keystroke('N', withMenu = true))
+
+    @throws(classOf[UserCancelException])
+    @throws(classOf[IOException])
+    override def action(): Unit = {
+      manager.offerSave()
+      manager.newModel()
+    }
+  }
+
+  class OpenAction(manager: FileManager, parent: Component)
+    extends ExceptionCatchingAction(I18N.gui.get("menu.file.open") + Ellipsis, parent) {
+    putValue(UserAction.ActionCategoryKey, UserAction.FileCategory)
+    putValue(UserAction.ActionGroupKey,    UserAction.FileOpenGroup)
+    putValue(UserAction.ActionRankKey,     Double.box(2))
+    putValue(ACCELERATOR_KEY,              UserAction.KeyBindings.keystroke('O', withMenu = true))
```

Comment:
Just updated. Thanks for the suggestion! As I mentioned, I had considered this, but it was good to get some feedback that this would actually be helpful.